design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +277,255 @@ permissionScope: Cluster
+ is installed into.
+
+At install time:
+ - Cluster Stack: generate a set of `ClusterRole` objects that will be aggregated to the environment
+  admin, edit, view `ClusterRole` objects.
+ - Namespaced Stack: generate a set of `ClusterRole` objects that will be aggregated to the namespace
+  specific admin, edit, view `ClusterRole` objects.
+
+#### Default Crossplane ClusterRoles
+
+The following built in roles will be installed and packaged with Crossplane, and updated with role aggregation as we
+ install additional stacks.
+
+The verbs supported in these roles(admin,edit,view) will model behavior based on existing logic around
+ [built-in kubernetes roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings).
+
+|ClusterRole Name | Permissions |
+|:---------------------|:----------|
+| crossplane-admin | Admin: ClusterStacks, Crossplane CRDs, Secrets in crossplane-system, Namespaces, (Cluster)RoleBindings, (Cluster)Roles |
+| crossplane-env-admin | Admin: ClusterStacks, globally scoped Crossplane CRDs, Secrets in crossplane-system. |
+| crossplane-env-edit | Edit: ClusterStacks, globally scoped Crossplane CRDs, Secrets in crossplane-system |
+| crossplane-env-view | View: Globally scoped Crossplane CRDs, Not permitted: secrets |
+
+We will provide a convenience group - crossplane:master - that binds to crossplane-admin with a
+ ClusterRoleBinding. This mirrors Kubernetes functionality for cluster-admin and system:master. At install time you will
+ then be able to impersonate a group that is bound to crossplane-admin.
+
+Example running a command as crossplane-admin
+```
+kubectl {any crossplane-admin operation} --as-group=crossplane:master
+```
+
+For discoverability of the roles for the environment we will add the following labels:
+ * `rbac.crossplane.io/role-environment: "true"`
+
+Match labels on these roles for auto-aggregation:
+ * `rbac.crossplane.io/aggregate-to-crossplane-admin: "truse"`
+
+#### ClusterRoles for a given app namespace.
+
+For any namespace that is annotated with `crossplane.io/managed-roles: true`, the stack-manager will install and manage
+ the following roles for that namespace that will be properly configured for ClusterRole aggregation as outlined below.
+ These roles should be bound to subjects to grant access to a particular namespace.
+
+ * `crossplane:ns:{ns-name}:admin`
+ * `crossplane:ns:{ns-name}:edit`
+ * `crossplane:ns:{ns-name}:view`
+
+For discoverability of the roles for a given namespace we will add the following labels:
+ * `rbac.crossplane.io/role-namespace: "true"`
+ * `{namespace-name}: "true"`
+
+Match Roles:
+ * `rbac.crossplane.io/aggregate-to-{install-scope}-{role}: "true"`
+ * `{namespace-name}: "true" // to match specific namespace.`
+
+#### ClusterRoles for a given (Cluster)Stack Install
+
+For a stack given stack version the stack-manager will create and or label a unique set of roles in the format
```

Comment:
This confused me at first because I thought the suggested format was how these roles would be labelled, but I think it's intended to be how they're named. It could be worth rewording to clarify.

Reply:
reworded.

For a stack given stack version the stack-manager will create a unique set of roles in the format stack:{repo-name}:{stack-name}:{version}:{role=system|admin|edit|view}.